### PR TITLE
feat: start new test from stash

### PR DESCRIPTION
**Pull request by ondrej-hosak**

### CODE REVIEW

File: lib/final-api/endpoint/ddtf.rb [hunk 1]
```diff
@@ -17,6 +17,42 @@ def registered(app)
             }.to_json
           end
 
+          app.get '/ddtf/tests/new' do
+            halt 400, {
+                status: 'error',
+                data: nil,
+                message: 'Please provide stashTSDLink link'
```
Comment (petrSchreiber) [inline lib/final-api/endpoint/ddtf.rb]: "link" should be replaced with "url parameter"

File: lib/final-api/endpoint/ddtf.rb [hunk 2]
```diff
@@ -17,6 +17,42 @@ def registered(app)
...
+              packageFrom: tsd['source']['git'] ? 'GIT' : 'UNC',
+              package: tsd['source']['git'] || tsd['source']['unc'],
+              strategy: tsd['defaultStrategy'],
+              build: nil,
+              description: tsd['description'],
+              scenarioScripts: false,
+              checkpoints: false,
+              stashTSD: params[:stashTSDLink],
+              runtimeConfigFields: tsd['runtimeConfig'],
+              tsdType: 'stash',
```
Comment (petrSchreiber) [inline lib/final-api/endpoint/ddtf.rb]: why is this filled, we ignore it

File: spec/integration/endpoint/ddtf_spec.rb [hunk 3]
```diff
@@ -275,6 +275,53 @@ def app
...
+      get '/ddtf/tests/new?stashTSDLink=link'
+      result = JSON.parse(last_response.body)
+      expect(last_response.status).to eq(200)
+
+      expect(result['packageFrom']).to eq 'GIT'
+      expect(result['stashTSD']).to eq 'link'
+      expect(result['package']).to eq 'some link'
+    end
+
+    context 'when Content fetcher throw error' do
```
Comment (petrSchreiber) [inline spec/integration/endpoint/ddtf_spec.rb]: throws

File: spec/integration/endpoint/ddtf_spec.rb [hunk 4]
```diff
@@ -275,6 +275,53 @@ def app
...
+        }.to_json
+      end
+    end
+
+    it 'returns 400 when missing stashTsdLink parameter' do
+      get '/ddtf/tests/new'
+      expect(last_response.status).to eq(400)
+    end
+
+    it 'returns new test payload' do
```
Comment (petrSchreiber) [inline spec/integration/endpoint/ddtf_spec.rb]: returns 200 and new test payload

File: lib/final-api/endpoint/ddtf.rb [hunk 5]
```diff
@@ -17,6 +17,42 @@ def registered(app)
...
+              halt 404, {
+                  status: 'error',
+                  data: nil,
+                  message: "Unable to load tsd from: #{params[:stashTSDLink]}"
+              }.to_json
+            end
+
+            halt 200, {
+              email: tsd['responsible'],
+              packageFrom: tsd['source']['git'] ? 'GIT' : 'UNC',
```
Comment (petrSchreiber) [inline lib/final-api/endpoint/ddtf.rb]: please consider using get_ikey for retrieving fields in case insensitive way

**Comment by ondrej-hosak** [comment]
@petrSchreiber @Trawais please review updated version

**Pull request merged**